src/tracker/tracker_sampler.jl

```diff
@@ -0,0 +1,98 @@
+abstract type AbstractROI end
+mutable struct BoxROI{T <: AbstractArray, S <: AbstractArray} <: AbstractROI
+    img::T
+    bound::S
+end
+
+abstract type AbstractTrackerSampler end
+mutable struct CurrentSample{F <: Float64, S <: Symbol} <: AbstractTrackerSampler
+    overlap::F
+    search_factor::F
+    mode::S
+    sampling_ROI::SVector{4, Integer}
+    valid_ROI::SVector{4, Integer}
```

Comment:
`sampling_region` and `valid_region` read better for me since I would guess this belonging to `AbstractROI`